src/core/server/saved_objects/routes/deprecations/delete_unknown_types.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IRouter } from '../../../http';
+import { catchAndReturnBoomErrors } from '../utils';
+import { deleteUnknownTypeObjects } from '../../deprecations';
+import { SavedObjectConfig } from '../../saved_objects_config';
+import { KibanaConfigType } from '../../../kibana_config';
+
+interface RouteDependencies {
+  config: SavedObjectConfig;
+  kibanaConfig: KibanaConfigType;
+  kibanaVersion: string;
+}
+
+export const registerDeleteUnknownTypesRoute = (
+  router: IRouter,
+  { config, kibanaConfig, kibanaVersion }: RouteDependencies
+) => {
+  router.post(
+    {
+      path: '/deprecations/_delete_unknown_types',
+      validate: false,
+    },
```

Comment:
A small security concern here: as we can't leverage the SOR to perform the cleanup, we're using the `internal` ES client instead (which also make sense, because we want to be able to delete all the unknown docs anyway, and because the SO security does not work with unknown types...).

However, this means that any authenticated user can potentially hit this endpoint, and cause the removal of the unknown docs. We could restrict the route with a `access:` tag, **however** this endpoint needs to be reachable by anyone having access to the UA management section to resolve the deprecation.

I don't think this is that big of a deal, but still worth a comment: are we fine with this?

Reply:
The likelihood is low that this gets abused and in most cases the impact will be low, it will probably just delete unused data. 

But as a pattern this feels wrong, it feels like something that will crop up for other teams too that have destructive correctiveActions e.g. https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/server/routes/deprecations.ts#L100

In reporting's case, security is enforced by Elasticsearch. So there's some precedent that just because you're able to use the upgrade assistant doesn't mean you'll automatically be able to perform all corrective actions.

I wonder if it wouldn't be fine to require the kibana_admin to perform these upgrade assistant fixes?

Reply:
> I wonder if it wouldn't be fine to require the kibana_admin to perform these upgrade assistant fixes?

we could use the scoped client instead of the internal one, that would do it, but then other users would still see the deprecation, and clicking the button would throw a server error... I don't really like it either to be honest.